src/rust/ide/ast/impl/src/lib.rs

```diff
@@ -1085,6 +1108,26 @@ impl Ast {
         Ast::from(infix)
     }
 
+    /// Creates an AST node with `TextLineFmt` shape.
+    pub fn text_line_fmt(text:Vec<SegmentFmt<Ast>>) -> Ast {
+        let text_line_fmt = TextLineFmt {text};
+        Ast::from(text_line_fmt)
+    }
+
+    /// Creates an AST node with `TextUnclosed` shape.
+    pub fn text_unclosed(line:TextLine<Ast>) -> Ast {
+        let text_unclosed = TextUnclosed {line};
+        Ast::from(text_unclosed)
+    }
+
+    /// Creates an AST node with `TextBlockFmt` shape.
+    pub fn text_block_fmt(text:Vec<TextBlockLine<SegmentFmt<Ast>>>) -> Ast {
+        let spaces = 0;
+        let offset = 0;
```

Comment:
Is this legal for a text block to have zero offset? 
(I have no idea, please confirm with Wojciech)

Reply:
I have exposed offset as a parameter.